Support OpenCV >=4

fglinka opened the pull request:
Building this task in Ubuntu 20.04 or later will fail, because the pkgconfig-file was renamed from `opencv.pc` to `opencv4.pc`. Thus I added some code that was written by Malte Wirkus for the orogen-linemod-module to fall back on opencv4 if the regular package-config file could not be found.

Further, CV_AA was moved to cv::LINE_AA in OpenCV 4 so I added a constant  that is set based on a `CV_MAJOR_VERSION` check.

I did not test this for opencv <4 though, as I got no distribution with that version on hand so maybe you want to do that before merging.

fglinka committed:
Support OpenCV >= 4

chhtz commented on apriltags.orogen:
```diff
@@ -3,7 +3,11 @@ version "0.1"
 using_library "apriltags"
 using_library "frame_helper"
-using_library "opencv"
+begin
+	using_library "opencv"
+rescue
+	using_library "opencv4"
+end
```
You can remove that entirely, since `frame_helper` already depends on opencv (and finds the correct version using CMake)

fglinka replied:
Okay, then I will get rid of that block later.

chhtz commented on tasks/Task.cpp:
```diff
@@ -4,6 +4,12 @@
 using namespace apriltags;
 
+#if CV_MAJOR_VERSION >= 4
+const int CV_LINE_TYPE = cv::LINE_AA;
```
`cv::LINE_AA` has been available at least since OpenCV3, so we don't need the `#if`-guards (I don't know about OpenCV2, but I don't think we still need to support that version).

fglinka replied:
I guess, in that case we could drop the CV_LINE_TYPE macro in general then and just use cv::LINE_AA directly in the code.

chhtz replied:
Yes, would also be fine for me.

fglinka committed:
Apply suggested review changes

This commit applies some changes suggested in
https://github.com/rock-perception/perception-orogen-apriltags/pull/19

- It removes the dependency to opencv from the manifest and orogen files
  as frame_helper will already take care of that dependency
- It removes CV_LINE_TYPE again and instead inserts cv::LINE_AA directly
  into the drawing calls as that enum value has been there since OpenCV
  3.0

chhtz commented:
Thanks for the PR!
